### PR TITLE
[DUOS-2933][risk=no] Update to use V4 algorithm

### PR DESCRIPTION
**Pull request by rushtong**
### Addresses
https://broadworkbench.atlassian.net/browse/DUOS-2933

### Summary
* Remove usages of commercial use
* Replace with non profit use where appropriate
* Use [V4 algorithm](https://github.com/DataBiosphere/consent-ontology/blob/develop/docs/Algorithms.md#version-4) fields for data use bucketing purposes
* See also this back end PR: https://github.com/DataBiosphere/consent/pull/2286

### Testing Notes
Testing this is a bit challenging as there are no NPU datasets in dev. I suggest the following steps:
1. Find a DAR that hits multiple datasets
2. Use the **PUT** `/api/dataset/{id}/datause` endpoint to modify your selected datasets with a data use that looks like this: `{ "generalUse": true, "nonProfitUse": true }`
3. Use the **POST** `/api/match/reprocess/purpose/{purposeId}` endpoint with your DAR reference id to reprocess the algorithm decisions for the new data uses on the modified datasets
4. View the dar collection in the admin or chairperson view to see the data use buckets. They should appropriately show NPU for the modified datasets.

----
Have you read [Terra's Contributing Guide](https://github.com/DataBiosphere/terra-ui/wiki/Contributor-Guide) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

**Review by aarohinadkarni — APPROVED** [review]
tested & looks good 👍

File: src/utils/BucketUtils.js [hunk 1]
```diff
@@ -353,44 +353,42 @@ const createRpVoteStructureFromBuckets = (buckets) => {
...
-    'aggregateResearch',
-    'collaboratorRequired',
-    'controlSetOption',
-    'ethicsApprovalRequired',
-    'gender',
-    'geneticStudiesOnly',
-    'geographicalRestrictions',
-    'illegalBehavior',
-    'manualReview',
+  const fields = [
```
Comment (fboulnois) [inline src/utils/BucketUtils.js:366]: Are we retiring `addiction`, `aggregateResearch`, `controlSetOption`, `manualReview`, and `otherRestrictions`?

**Review by fboulnois — APPROVED** [review]
One questions, otherwise looks good:

File: src/utils/BucketUtils.js (hunk 1 above)
Comment (rushtong) [reply src/utils/BucketUtils.js:366]: Yes, we are - they are no longer part of the data use schema

**Pull request merged**